walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -89,11 +61,69 @@ private void addRangeToResults(int validRangeStart, int validRangeEnd,
       }
     }
 
-    // Add last block of day to results (if long enough)
+    // Add last block of day to validRanges (if long enough)
     if (minInvalidRangeEnd < TimeRange.END_OF_DAY) {
-      addRangeToResults(minInvalidRangeEnd, TimeRange.END_OF_DAY, true, results, duration);
+      addRange(minInvalidRangeEnd, TimeRange.END_OF_DAY, true, validRanges, duration);
     }
 
-    return results;
+    return validRanges;
+  }
+
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    Collection<TimeRange> validRanges = new ArrayList<TimeRange>();
+    Collection<String> attendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    long duration = request.getDuration();
+
+    // Not possible to have an event longer than a full day
+    if (duration > TimeRange.WHOLE_DAY.duration()) {
+      return validRanges;
+    }
+
+    if (attendees.isEmpty()) {
+      if (optionalAttendees.isEmpty()) {
+        // If no one's attending, there are no conflicts
+        validRanges.add(TimeRange.WHOLE_DAY);
+        return validRanges; 
+      } else { // All attendees optional
+        attendees = optionalAttendees;
+        optionalAttendees = attendees; // empty set
```

Comment:
I don't think that optionalAttendees will be the empty set after you execute this. I guess it doesn't matter, but since you point optionalAttendees to attendees, which points to optionalAttendees, line 90 doesn't change anything.